### DISCUSSION TITLE
S21/S403 Current Output? MD8942 Replacement to Support Additional MCU for New Capabilities (mmWave) via USB

### DISCUSSION THREAD

**Discussion by bai-yi-bai** [revk/xESP32-Faikout#353] [Ideas]
Hello,

Are there any available specifications about the S21 or S403's maximum current output on the 5 volt and 12 volt rails?
I read that the MD8942 step-down converter the Faikin uses provides about 600 mA of power which I assume is mostly consumed by the ESP32 and the RGB LED.

I ask because I would like to add another microcontroller to support additional sensors. For example, I'd like to mount a mmWave presence detector. I suppose one option is to "remix" the Faikin to provide a stronger step down converter which provides more current, along with a USB-C or USB A female connector, which would allow users to connect commercial sensors or DIY ESP32s running ESPHome or Tasmota. The concern I have is about how much current the Daikin air conditioners provide on the S21/S403 ports. Of course, discussions like these should also consider the maximum current rating of the crimp connectors and wiring. EH connectors are rated for a maximum of 3 A.

I am extremely satisfied with the 4 Faikins I deployed. MQTT and the ability to adjust the temperature in half degree steps makes using these units so much better. I am now looking into expanding my smart home, especially with presence detection to trigger more automation routines, for example to trigger lights or to turn off air conditioners if there is no one in the room. The problem I have is that I live in an apartment made of concrete and steel, which makes setting up additional sensors a bit difficult. My Daikin split air conditioners have power and a prominent position in each room. I would like to incorporate mmWave presence detectors.  I think currently the Faikin uses both of the UART connectors, which rule out incorporating a mmWave sensor such as the HLK-LD1125H-24g, plus that would take a lot of extra effort to add the code to support it.

Here are some YouTubers who have created videos about mmWave:
https://www.youtube.com/watch?v=Kt1FpRM8R18&ab_channel=MakeItWork
https://www.youtube.com/watch?v=s-GzUTyIH9c&ab_channel=AndreasSpiess

**Comment by revk** [comment]
No idea, maybe someone else knows.

**Reply by bai-yi-bai** [reply]
Thank you. No problem. 

My new plan is to splice in an LM2526 DC-DC converter to the 12 volt S21/S403 rail and ground. Conveniently, the modules I have on hand have USB female connectors. To anyone reading this, please be careful. I am taking on the full risk that I might be exceeding the designed power circuit limits. ~500 mA to power another ESP32 dev board + mmWave sensor hopefully doesn't break anything. I'd rather do this than try to wire an electrical outlet to the ~220 volt AC.

I found a comment you posted which was also helpful.
https://github.com/revk/ESP32-Faikin/issues/206#issuecomment-1912990331

> The power for the Faikin is on the "12V" (which may be 14V even). It actually works with anything from 5V to 35V.
> 
> The 5V connection (pin 1) is for a pull up. It does not power the Faikin board. If you try to power the Faikin from this it is likely to not have enough current (hence upsetting the aircon if you do that).

**Comment by dzungpv** [comment]
> Hello,
> 
> Are there any available specifications about the S21 or S403's maximum current output on the 5 volt and 12 volt rails? I read that the MD8942 step-down converter the Faikin uses provides about 600 mA of power which I assume is mostly consumed by the ESP32 and the RGB LED.
> 
> I ask because I would like to add another microcontroller to support additional sensors. For example, I'd like to mount a mmWave presence detector. I suppose one option is to "remix" the Faikin to provide a stronger step down converter which provides more current, along with a USB-C or USB A female connector, which would allow users to connect commercial sensors or DIY ESP32s running ESPHome or Tasmota. The concern I have is about how much current the Daikin air conditioners provide on the S21/S403 ports. Of course, discussions like these should also consider the maximum current rating of the crimp connectors and wiring. EH connectors are rated for a maximum of 3 A.
> 
> I am extremely satisfied with the 4 Faikins I deployed. MQTT and the ability to adjust the temperature in half degree steps makes using these units so much better. I am now looking into expanding my smart home, especially with presence detection to trigger more automation routines, for example to trigger lights or to turn off air conditioners if there is no one in the room. The problem I have is that I live in an apartment made of concrete and steel, which makes setting up additional sensors a bit difficult. My Daikin split air conditioners have power and a prominent position in each room. I would like to incorporate mmWave presence detectors. I think currently the Faikin uses both of the UART connectors, which rule out incorporating a mmWave sensor such as the HLK-LD1125H-24g, plus that would take a lot of extra effort to add the code to support it.
> 
> Here are some YouTubers who have created videos about mmWave: https://www.youtube.com/watch?v=Kt1FpRM8R18&ab_channel=MakeItWork https://www.youtube.com/watch?v=s-GzUTyIH9c&ab_channel=AndreasSpiess
I get from the user manual:
 - BRP069C81 with X50a and X35a power port: Power supply: DC 14 V/ 16 V / 100 mA
 - BRP069C51 with S21 power by 14V pin: Power supply: DC 14 V / 100 mA
